src/lib/model/OutboundRequestToPayTransferModel.js

```diff
@@ -131,19 +148,31 @@ class OutboundRequestToPayTransferModel {
                     break;
 
                 case 'quoteReceived':
-                    // next transition is requestOTP
-                    await this.stateMachine.requestOTP();
-                    if(this.data.initiatorType !== 'BUSINESS') {
-                        this._logger.log(`OTP received for transactionId: ${this.data.requestToPayTransactionId} and transferId: ${this.data.transferId}`);
-                        if(this.stateMachine.state === 'otpReceived' && !this._autoAcceptR2PDeviceOTP) {
-                            //we break execution here and return the otp response details to allow asynchronous accept or reject
-                            //of the quote
-                            await this._save();
-                            return this.getResponse();
+                    // decide PISP or OTP flow
+                    if (this._enablePISPMode) {
+                        await this.stateMachine.requestAuthorization();
+                        await this._save();
+                        // let executeTransfer in recursive call of run()
```

Comment:
@eoln is there a need to return or break here?

Reply:
the break is in L168 elsewhere we need a break to have a recursive call of run()